sdk/provisioning/Azure.Provisioning.CloudMachine/src/CloudMachineInfrastructure.cs

```diff
@@ -45,43 +45,46 @@ public class CloudMachineInfrastructure : Infrastructure
     ///// </summary>
     //public BicepParameter PrincipalNameParameter => new BicepParameter("principalName", typeof(string));
 
-    public CloudMachineInfrastructure(string name = "cm") : base(name!)
+    public CloudMachineInfrastructure(string cmid) : base("cm")
```

Comment:
shouldn't we pass `cmid` to the base also?

Reply:
Yeah, `base(name)` is only used for the generated file name (i.e., this will always result in a `cm.bicep`).  You probably want to keep that constant for usability's sake.  You could potentially add a second .ctor overload in the future if you wanted to make it easily customizable.